backend/src/models/requestsModel.ts

```diff
@@ -265,6 +265,35 @@ export default {
     }
     request.approvers = request.approvers ?? [];
 
+    const room = await db.room.findUnique({ where : { roomName : request.roomName } })
+    const pendingUser = await db.request.findMany({
+      where: {
+        authorUtorid: user.utorid,
+        status: RequestStatus.pending,
+        roomName: request.roomName,
+      }
+    })
```

Comment:
This info can be fetched when fetching this user, try limiting the number of DB queries